### ISSUE TITLE
java.io.FileNotFoundException application.properties

### ISSUE THREAD

**Issue by u84six** [PopMedNet-Team/Mobile-Application#23]
In the documentation for setting up WCP, it says to put the application.properties file in system/server. I'm not sure what that means. I configured it and put it under a directory that I created: /usr/local/hphcServers/application.properties

Then in the tomcat home directory, I configured context.xml:

<Context>

    
    
    <WatchedResource>WEB-INF/web.xml</WatchedResource>
    <WatchedResource>${catalina.base}/conf/web.xml</WatchedResource>

    
    
    <Parameter name="property_file_location_prop" value="/usr/local/hphcServers/" override="1"/>
    <Parameter name="property_file_name" value="application" override="1"/>
    <Parameter name="property_file_location_config" value="/usr/local/hphcServers/application.properties" override="1"/>
    <Parameter name="property_file_location_path" value=" /usr/local/hphcServers/application.properties" override="1"/>
</Context>

When I restart tomcat, the catalina.out, fdahpStudyDesigner.log, and studyMetaDataWS.log says that it can't find the file:

"Could not open ServletContext resource [/usr/local/hphcServers/application.properties]"

Where should this file go?

**Comment by u84six** [comment]
I tried moving the file under CATALINA_HOME/webapps and changing the context.xml to reflect the new path and that didn't solve the problem. The whole directory has tomcat group r/w access.

**Comment by seanthingee** [comment]
@u84six I had this same issue and after some fiddling I happened upon a workaround.

I moved application.properties to tomcat's CATALINA_HOME/webapps/fdahpStudyDesigner/.

And changed the path in context.xml, like this:
```
<Parameter name="property_file_location_prop" value="/usr/local/" override="1"/>
<Parameter name="property_file_name" value="application" override="1"/>
<Parameter name="property_file_location_config" value="application.properties" override="1"/>
<Parameter name="property_file_location_path" value=" application.properties" override="1"/>
```

(And I guess I left the /usr/local in there and it still worked...)

**Comment by u84six** [comment]
> @u84six I had this same issue and after some fiddling I happened upon a workaround.
> 
> I moved application.properties to tomcat's CATALINA_HOME/webapps/fdahpStudyDesigner/.
> 
> And changed the path in context.xml, like this:
> 
> ```
> <Parameter name="property_file_location_prop" value="/usr/local/" override="1"/>
> <Parameter name="property_file_name" value="application" override="1"/>
> <Parameter name="property_file_location_config" value="application.properties" override="1"/>
> <Parameter name="property_file_location_path" value=" application.properties" override="1"/>
> ```
> 
> (And I guess I left the /usr/local in there and it still worked...)

@seanthingee thanks for the info, I'll give it a try. Did you also manually move the war files into webapps directory? The documentation says to do that, but I've noticed that the maven build deploys as well.

**Comment by u84six** [comment]
@seanthingee Also, did you do the same thing for WCP-WS?

**Comment by u84six** [comment]
@seanthingee so after I moved the war files into webapps and restarted the server, tomcat is not creating the directory for fdahpStudyDesigner, only the one for StudyMetaData-0.0.1-SNAPSHOT.

What did you do to get the directory into webapps?

**Comment by u84six** [comment]
> @seanthingee so after I moved the war files into webapps and restarted the server, tomcat is not creating the directory for fdahpStudyDesigner, only the one for StudyMetaData-0.0.1-SNAPSHOT.
> 
> What did you do to get the directory into webapps?

@seanthingee never mind about this one. It just took tomcat a while to create fdahpStudyDesigner directory.

**Comment by seanthingee** [comment]
> Also, did you do the same thing for WCP-WS?

Honestly I don't know what WCP-WS is. The Setup instructions doesn't seem to mention it. Only WCP.

**Comment by seanthingee** [comment]
Oh I see it now, a very brief mention mixed in with studyMetaData. This documentation is not great.

**Comment by u84six** [comment]
@seanthingee yeah, the documentation is not good, and it doesn't seem that the devs checking in changes ever address issues here.

**Comment by u84six** [comment]
@seanthingee putting the application.properties file in fdahpStudyDesigner did not fix the problem for me. Still getting the same errors that it can't find the file. It's actually worse now, which seems rather odd. So it tells me that some parts of their code did find the file, while others did not. Ugh

**Comment by seanthingee** [comment]
@u84six Another thing to try, which I had working first before going back to using the context.xml file, is to update CATALINA_HOME/webapps/fdahpStudyDesigner/WEB-INF/ApplicationContext.xml. Find the line that has `location="#{contextParameters.property_file_location_config}" />` and change it to `location="application.properties" />`.

**Comment by u84six** [comment]
@seanthingee ok, I tried this, but it seems like it made it worse because I'm seeing other errors associated with database drivers not making a connection, and the db urls are in application.properties. So this tells me that when I put the static path in the context.xml, part of their code finds it and other parts don't. Ugh

**Comment by kuttima** [comment]
I am having the same issue. Where you guys able to resolve this?

**Comment by FDA-MyStudies** [comment]
Hi @u84six ,

Thanks for your interest in the MyStudies system.  Apologies for the delay in response.  Below are a few clarifications

- Remove one space, file path included one space in starting, line #4 (value=" /usr/...)

- application.properties file can be kept anywhere, same path should be defined in tomcat context.xml file

Example, its my local machine

<Parameter name="property_file_location_prop" value="/usr/local/" override="1"/>
<Parameter name="property_file_name" value="application" override="1"/>
<Parameter name="property_file_location_config" value="file:/usr/local/application.properties" override="1"/>
<Parameter name="property_file_location_path" value="/usr/local/application.properties" override="1"/>

my application.properties is located in /usr/local/application.properties path.

- WCP-WS is the WCP webservices, all WCP content are served to mobile application and response server through this webservice project, as a part of WCP setup, you need to setup the webservice project as well. Our setup document is explained about this, please refer it.

https://github.com/PopMedNet-Team/FDA-My-Studies-Mobile-Application-System/blob/2019.10/FDAMyStudies_Release%202019.10_SetupInstructions_V1.0.pdf

- Rename the StudyMetaData-0.0.1-SNAPSHOT to StudyMetaData.war, refer the section 2.4 in setup document.

- 'database drivers not making a connection' : Looking at this message, looks like you are able to provide the proper path of application.properites file in context file, as a next step, you need to establish the proper database configuration. Check your database configuration parameters."

**Comment by liujoshua** [comment]
Spent several hours figuring this out. The format of `property_file_location_config` and `property_file_location_path` are different.

Here's what worked for me:
```
<Context>
	<!-- Used by ApplicationContext.xml -->
	<Parameter name="property_file_location_config" value="file:///usr/local/tomcat/config/application.properties" override="1"/>
	<!-- Used by FdahpStudyDesignerUtil.java -->
	<Parameter name="property_file_location_path" value="/usr/local/tomcat/config/application.properties" override="1"/>
</Context>
